src/AutoBinder.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MichaelRubel\AutoBinder;
+
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+use MichaelRubel\AutoBinder\Traits\BindsToContainer;
+
+class AutoBinder
+{
+    use BindsToContainer;
+
+    /**
+     * Base class namespace.
+     *
+     * @var string
+     */
+    public string $classNamespace = 'App';
+
+    /**
+     * Target class folder.
+     *
+     * @var string
+     */
+    public string $classFolder = 'Services';
+
+    /**
+     * Interface namespace (can be fully qualified).
+     *
+     * @var string
+     */
+    public string $interfaceNamespace = 'Interfaces';
+
+    /**
+     * Postfix convention for interfaces.
+     *
+     * @var string
+     */
+    public string $interfaceNaming = 'Interface';
+
+    /**
+     * Base class folder.
+     *
+     * @var string
+     */
+    public string $basePath = 'app';
+
+    /**
+     * The type of bindings.
+     *
+     * @var string
+     */
+    public string $bindingType = 'bind';
+
+    /**
+     * When the class name is met, these dependencies are passed to the concrete.
+     *
+     * @var array
+     */
+    public array $dependencies = [];
+
+    /**
+     * Subdirectories to ignore when scanning.
+     *
+     * @var array
+     */
+    public array $excludesFolders = [];
+
+    /**
+     * Assign a new class folder.
+     *
+     * @param string|null $classFolder
+     */
+    final public function __construct(?string $classFolder = null)
+    {
+        if ($classFolder) {
+            $this->classFolder = $classFolder;
+        }
+    }
+
+    /**
+     * Create the object with target folder assigned.
+     *
+     * @param string|array $folder
+     *
+     * @return static|Collection
+     */
+    public static function from(string|array $folder): static|Collection
+    {
+        if (func_num_args() > 1) {
+            $folders = is_array($folder) ? $folder : func_get_args();
```

Comment:
refactoring ;)
```suggestion
            $folders = is_array($folder)
            ? $folder
            : func_get_args();
```

Reply:
Thanks, but I prefer to stick with a one-liner. 🙂